libvast/vast/chunk.hpp

```diff
@@ -78,11 +78,18 @@ class chunk final : public caf::ref_counted {
   /// accidental copies when calling this function.
   /// @returns A chunk pointer or `nullptr` on failure.
   template <class Buffer>
-    requires(!(std::is_lvalue_reference_v<
-                 Buffer> || std::is_trivially_move_assignable_v<Buffer>))
-  static auto make(Buffer&& buffer) -> decltype(as_bytes(buffer), chunk_ptr{}) {
-    const auto view = as_bytes(buffer);
-    return make(view, [buffer = std::exchange(buffer, {})]() noexcept {
+    requires(!std::is_lvalue_reference_v<Buffer> &&          //
+             !std::is_trivially_move_assignable_v<Buffer> && //
+             requires(const Buffer& buffer) {
+               { as_bytes(buffer) } -> concepts::same_as<view_type>;
+             })
+  static auto make(Buffer&& buffer) -> chunk_ptr {
+    // Move the buffer into a unique pointer; otherwise, we might run into
+    // issues with small buffer optimizations where the view no longer points
+    // to the correct data once we moved the buffer into the deleter.
```

Comment:
The comment seems to be too narrow, even without small buffer optimization there is no requirement that a move constructor must preserve all internal memory addresses.

Reply:
I reworded the comment in a separate commit.